src/store/actions.js

```diff
@@ -1,16 +1,43 @@
 import Vue from 'vue';
 import Config from '../config';
 
+const updateStateWithNewProgress = (commit, progress) => {
+    const { stocks, funds, stockPortfolio } = progress;
+    const portfolio = { funds, stockPortfolio };
+
+    commit('SET_STOCKS', stocks);
+    commit('SET_PORTFOLIO', portfolio);
+};
+
 export const importProgress = ({ commit }) => {
     Vue.http.get(Config.firebase.progressEndpoint)
         .then(response => response.json())
         .then(progress => {
             if (progress) {
-                const { stocks, funds, stockPortfolio } = progress;
-                const portfolio = { funds, stockPortfolio };
-
-                commit('SET_STOCKS', stocks);
-                commit('SET_PORTFOLIO', portfolio);
+                updateStateWithNewProgress(commit, progress);
+                alert('Your progress was successfully imported.');
             }
+        })
+        .catch(error => {
+            alert('There is a problem importing your progress. Probably, the last exported progress was corrupted.');
+            console.log("Error while importing progress: ", error);
         });
+};
+
+export const loadProgress = ({ commit }) => {
+    const rawProgress = localStorage.getItem('progress');
```

Comment:
Probably, would be better to use a constant for this purpose, in order to avoid problems in the future.... 🤔